client/src/services/phases.js

```diff
@@ -0,0 +1,89 @@
+import store from 'store';
+import { API_URL } from '../constants';
+
+/**
+ *
+ * @param {*} response API Request response
+ * @param {*} columns Which columns we're expected to map to
+ * @returns
+ */
+const mapPhasesResponse = async (data, columns) => {
+  const rowsData = data.map((row) => {
+    // Pull all relevant props from row based on columns constant
+    const mappedRow = columns.reduce(
+      (accumulator, column) => ({
+        ...accumulator,
+        [column.id]: row[column.id],
+      }),
+      {}
+    );
+    // Add additional props (user ID, button) to row
+    return {
+      ...mappedRow,
+      id: row.id,
+    };
+  });
+  return rowsData;
+};
+
+export const fetchPhases = async () => {
+  const response = await fetch(`${API_URL}/api/v1/phase-allocation/`, {
+    headers: { Authorization: `Bearer ${store.get('TOKEN')}` },
+    method: 'GET',
+  });
+
+  if (response.ok) {
+    const { data } = await response.json();
+    return data;
+  }
+  throw response;
+};
+
+export const FetchMappedPhases = async (columns) => {
+  try {
+    const data = await fetchPhases();
+    return mapPhasesResponse(data, columns);
+  } catch (error) {
+    console.warn('Failed to get phases!', error);
```

Comment:
How can we get this beyond a console.warn to actually displaying an error? It doesn't necessarily have to be in this PR, but we should consider it

Reply:
We have toast notifications elsewhere already - ideally I would want to see errors like this propagated to the component calling it, and then displayed on-page. This solution is more in line with how other similar tables are handled, though, so I agree with saving it for another PR.